docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,425 @@
+
+## Transparent upgradable proxies
+
+Transparent upgradable proxies provide a way to upgrade a smart contract without changing its address or requiring any change in the contract's interaction code. With transparent proxies, a contract's address is owned by a proxy contract, which forwards all calls to the actual contract implementation. When a new implementation is deployed, the proxy can be upgraded to point to the new implementation, allowing for seamless upgrades without requiring changes to the contract's interaction code.
+
+To deploy a simple upgradable contract on zkSync Era local setup, first create a test wallet and add it to the new Deployer.
+
+```typescript
+  const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+  const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+  const deployer = new Deployer(hre, zkWallet);
+```
+
+After that, load the `Box` artifact and call the `deployProxy` method from the `zkUpgrades` hre property.
+
+```typescript
+  const contractName = 'Box';
+  const contract = await deployer.loadArtifact(contractName);
+  await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+```
+
+The `deployProxy` method deploys your implementation contract on zkSync Era, deploys the proxy admin contract, and finally, deploys the transparent proxy.
+
+### Full deploy proxy script
+
+```typescript
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { Wallet } from 'zksync-web3';
+
+const hre = require('hardhat');
+
+async function main() {
+    const contractName = 'Box';
+    console.log('Deploying ' + contractName + '...');
+
+    const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+    const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+    const deployer = new Deployer(hre, zkWallet);
+
+    const contract = await deployer.loadArtifact(contractName);
+    const box = await hre.zkUpgrades.deployProxy(deployer.zkWallet, contract, [42], { initializer: 'store' });
+
+    await box.deployed();
+    console.log(contractName + ' deployed to:', box.address);
+
+    box.connect(zkWallet);
+    const value = await box.retrieve();
+    console.log('Box value is: ', value.toNumber());
+}
+
+main();
+```
+
+### Hardhat config
+
+:::warning
+
+- deployProxy method (and other deploy/upgrade methods from the zkUpgrades) needs to know which wallet to use to deploy smart contracts.
+- For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network.
+- This provider is configured in the hardhat config file, by stating the RPC url of the network to connect to.
+:::
```

Comment:
```suggestion
:::warning

- deployProxy method (and other deploy/upgrade methods from the zkUpgrades) needs to know which wallet to use to deploy smart contracts.
- For this reason, the wallet needs to have a configured provider that connects it to the specific zkSync network.
- This provider is configured in the hardhat config file, by stating the RPC url of the network to connect to.
:::

### Hardhat config

```

Reply:
this was a pending comment from https://github.com/matter-labs/zksync-web-era-docs/pull/422